### PR TITLE
Do not panic when reading public key of relay

### PR DESCRIPTION
**Pull request by MarkusPettersson98**
Do not panic when getting public key of relay due to unwrapping an unexpected relay type. The unwrap happened if an OpenVPN relay was selected on Android, which should not happen in the first place.

- - -
This change is [Reviewable](https://reviewable.io/reviews/mullvad/mullvadvpn-app/6033)

### CODE REVIEW

File: mullvad-daemon/src/tunnel.rs [hunk 1]
```diff
@@ -44,11 +44,8 @@ pub enum Error {
     #[error("Not logged in on a valid device")]
     NoAuthDetails,
 
-    #[error("No relay available")]
-    NoRelayAvailable,
-
-    #[error("No bridge available")]
-    NoBridgeAvailable,
+    #[error("Failed to select a mathcing relay")]
```
Comment (MrChocolatine) [inline mullvad-daemon/src/tunnel.rs:47]:
```suggestion
    #[error("Failed to select a matching relay")]
```

**Review by dlon — COMMENTED** [review]
> Reviewed 5 of 5 files at r1, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/mullvad/mullvadvpn-app/6033)* status: all files reviewed, 2 unresolved discussions (waiting on @MarkusPettersson98)

___
*[`mullvad-relay-selector/src/relay_selector/detailer.rs` line 180 at r1](https://reviewable.io/reviews/mullvad/mullvadvpn-app/6033#-Nu-QlomCBCdXAGEtf6F:-Nu-QlomCBCdXAGEtf6G:bb7462z) ([raw file](https://github.com/mullvad/mullvadvpn-app/blob/530971a69dd19f49edefec31c01ea71d09a5531c/mullvad-relay-selector/src/relay_selector/detailer.rs#L180)):*
> ```XML
> }
> 
> /// TODO: Document
> ```

TODO

File: mullvad-daemon/src/tunnel.rs (hunk 1 above)
Comment (MarkusPettersson98) [reply mullvad-daemon/src/tunnel.rs]: Thanks!

**Review by dlon — APPROVED** [review]
> Reviewed 3 of 3 files at r2, 2 of 2 files at r3, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/mullvad/mullvadvpn-app/6033)* status: all files reviewed, 1 unresolved discussion (waiting on @MrChocolatine)

**Review by dlon — APPROVED** [review]
> Reviewed 2 of 2 files at r4, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/mullvad/mullvadvpn-app/6033)* status: all files reviewed, 1 unresolved discussion (waiting on @MrChocolatine)

**Pull request merged**